packages/components/src/dateInput/DateInput.js

```diff
@@ -2,58 +2,107 @@
 
 import React from 'react';
 import { defaultTokens } from '@kiwicom/orbit-design-tokens';
-import { format, setMonth } from 'date-fns';
 import { View } from 'react-native';
 import {
   StyleSheet,
   Picker,
   Text,
   TextInput,
 } from '@kiwicom/universal-components';
+import { setMonth, getDate, getMonth, getYear, format } from 'date-fns';
+import { MAX_YEAR_LENGTH, MAX_DAY_LENGTH } from '@kiwicom/margarita-config';
+
+import { composeDateFromStrings } from './helpers';
 
 type Props = {|
-  +onDateChange: (string, string) => void,
-  +date: {
-    +day: string,
-    +month: string,
-    +year: string,
-  },
-  +errors: {
-    +day: string,
-    +year: string,
-  },
+  +onDateChange: Date => void,
+  +date: ?Date,
+  +editMode: boolean,
+  +getDateRequest: boolean,
 |};
 
 type Month = {|
   value: string,
   label: string,
 |};
 
-class DateInput extends React.Component<Props> {
-  getMonthOptions = () => {
-    return [...Array(12).keys()].map<Month>(
-      (currentMonth: number): Month => ({
-        value: String(currentMonth),
-        label: format(setMonth(new Date(), currentMonth), 'MMMM'),
-      }),
-    );
+type State = {|
+  day: string,
+  month: string,
+  year: string,
+  hasPrefilledState: boolean,
+|};
+
+const getMonthOptions = () => {
+  return [...Array(12).keys()].map<Month>(
+    (currentMonth: number): Month => ({
+      value: String(currentMonth),
+      label: format(setMonth(new Date(), currentMonth), 'MMMM'),
+    }),
+  );
+};
+
+const initialState = {
+  day: '',
+  month: '0',
+  year: '',
+  hasPrefilledState: false,
+};
+
+export default class DateInput extends React.Component<Props, State> {
+  state = {
+    ...initialState,
```

Comment:
Why not to add it directly? You use initialState only once.

Reply:
Agree, will remove it